static/js/components/NewInstrument.jsx

```diff
@@ -0,0 +1,102 @@
+import React from "react";
+import { useDispatch, useSelector } from "react-redux";
+import Form from "@rjsf/material-ui";
+import { showNotification } from "baselayer/components/Notifications";
+import { submitInstrument } from "../ducks/instrument";
+
+const NewInstrument = () => {
+  const { instrumentList } = useSelector((state) => state.instruments);
+  const { telescopeList } = useSelector((state) => state.telescopes);
+  const dispatch = useDispatch();
+
+  const handleSubmit = async ({ formData }) => {
+    if (formData.group_id === -1) {
+      delete formData.group_id;
+    }
+    const result = await dispatch(submitInstrument(formData));
+    if (result.status === "success") {
+      dispatch(showNotification("Instrument saved"));
+    }
+  };
+
+  const api_classnames = [{ enum: [null], title: "No API" }];
+  instrumentList?.forEach((instrument) => {
+    if (instrument.api_classname) {
+      api_classnames.push({
+        enum: [instrument.api_classname],
+        title: instrument.api_classname,
+      });
+    }
+  });
+
+  const filters = [];
+  instrumentList?.forEach((instrument) => {
+    instrument.filters?.forEach((filter) => {
+      filters.push(filter);
+    });
+  });
+  const filtersUnique = [...new Set(filters)];
+
+  const uiSchema = {
+    filters: {
+      "ui:widget": "checkboxes",
```

Comment:
can we make this two-columns? If we get a few more filters this form becomes too tall.

Reply:
https://stackoverflow.com/questions/64529750/multi-column-support-for-react-jsonschema-form
surprisingly hard request... what do you think?

Reply:
If it is not easy to do I'd advocate skipping it for now. If the filter list grows too long we can figure out how to change it then.